src/server.cpp

```diff
@@ -3289,9 +3289,11 @@ bool Server::hudSetFlags(RemotePlayer *player, u32 flags, u32 mask)
 	if (!player)
 		return false;
 
-	SendHUDSetFlags(player->getPeerId(), flags, mask);
-	player->hud_flags &= ~mask;
-	player->hud_flags |= flags;
+	u32 new_hud_flags = (player->hud_flags & ~mask) | flags;
+	if (new_hud_flags != player->hud_flags) {
+		SendHUDSetFlags(player->getPeerId(), flags, mask);
+		player->hud_flags = new_hud_flags;
+	}
```

Comment:
Why not put the code below in the if too?

Reply:
or `return true;` when `new_hud_flags == player->hud_flags`

Reply:
I mostly cared about the networking so this was overly narrow. That would change behavior though: If the flags don't change and the `playersao` is somehow `null`, this would now return `true` instead of `false`. Is that what we want? I can't tell, I'm not familiar with MT's codebase. I will make the desired changes and you can revert the last commit should you decide against it.